sdk/storage/storage-file-share/src/ShareFileClient.ts

```diff
@@ -1651,18 +1651,88 @@ export class ShareFileClient extends StorageClient {
    * @param {number} offset From which position of the Azure File to download
    * @param {number} [count] How much data to be downloaded. Will download to the end when passing undefined
    * @param {FileDownloadToBufferOptions} [options]
-   * @returns {Promise<void>}
+   * @returns {Promise<Buffer>}
    */
   public async downloadToBuffer(
     buffer: Buffer,
-    offset: number = 0,
+    offset?: number,
     count?: number,
-    options: FileDownloadToBufferOptions = {}
-  ): Promise<void> {
+    options?: FileDownloadToBufferOptions
+  ): Promise<Buffer>;
+
+  /**
+   * ONLY AVAILABLE IN NODE.JS RUNTIME
+   *
+   * Downloads an Azure file in parallel to a buffer.
+   * Offset and count are optional, pass 0 for both to download the entire file
+   *
+   * @param {number} offset From which position of the Azure file to download
+   * @param {number} [count] How much data to be downloaded. Will download to the end when passing undefined
+   * @param {FileDownloadToBufferOptions} [options]
+   * @returns {Promise<Buffer>}
+   */
+  public async downloadToBuffer(
+    offset?: number,
+    count?: number,
+    options?: FileDownloadToBufferOptions
+  ): Promise<Buffer>;
+
+  public async downloadToBuffer(
+    bufferOrOffset?: Buffer | number,
+    offsetOrCount?: number,
+    countOrOptions?: FileDownloadToBufferOptions | number,
+    optOptions: FileDownloadToBufferOptions = {}
+  ): Promise<Buffer> {
+    let buffer: Buffer | undefined = undefined;
+    let offset: number;
+    let count: number;
+    let options: FileDownloadToBufferOptions = optOptions;
+
+    if (bufferOrOffset instanceof Buffer) {
+      buffer = bufferOrOffset;
+      offset = offsetOrCount || 0;
+      count = typeof countOrOptions === "number" ? countOrOptions : 0;
+    } else {
+      offset = typeof bufferOrOffset === "number" ? bufferOrOffset : 0;
+      count = typeof offsetOrCount === "number" ? offsetOrCount : 0;
+      options = (countOrOptions as FileDownloadToBufferOptions) || {};
+    }
+
     const { span, spanOptions } = createSpan(
       "ShareFileClient-downloadToBuffer",
       options.tracingOptions
     );
+
+    if (!count) {
+      // User did not specify the bytes, so we need to ask the service how large the file is
+      const response = await this.getProperties({
```

Comment:
This is present in the try block below.
https://github.com/Azure/azure-sdk-for-js/pull/6084/files?file-filters%5B%5D=.md&file-filters%5B%5D=.ts#diff-458825853ee90eb4621dbbff6b5955f9R1761

Please update accordingly. 
(Move everything after the initialization of `{ span, spanOptions }` into the try block that is already present below.)

Reply:
@HarshaNalluru Can you provide a pointer to how we did this in blobs for reference?

Reply:
This PR adds two if blocks just before the existing try block.
Current PR -
```js
const { span, spanOptions }
if (!count) { --- }           - newly added
if (!buffer) { --- }          - newly added
try{ <old code> }
```
Suggesting to 
- Remove `if (!count) { --- }` block which is already present in the `try{ <old code> }`.
- And move the `if (!buffer) { --- }` inside the `try{ <old code> }` 
put it right after the existing `if (!count) { --- }`(the place where getProperties() is invoked and count is initialized.)

Reply:
*Like this - https://github.com/Azure/azure-sdk-for-js/pull/5671/files?file-filters%5B%5D=.ts#diff-f00e773677816decb47917ac5c2af667L1487-R1572